tensorflow/stream_executor/gpu/gpu_types.h

```diff
@@ -20,6 +20,7 @@ limitations under the License.
 
 #if TENSORFLOW_USE_ROCM
 
+#define __HIP_DISABLE_CPP_FUNCTIONS__
```

Comment:
My assumption was that this macro would be added as a default cxx_opt for files compiled with hipcc (just to be on the safe / paranoid side)

Reply:
I'd like to understand how things work in GPU common runtime in upstream TF. In case everything moves away from linking with HIP / CUDA runtime, we might want to consider setting it in `cxx_opt`.